### ISSUE TITLE
Hub application for Windows IoT Core

### ISSUE THREAD

**Issue by hansmbakker** [getsenic/nuimo-windows#17]
Hi,

I created the start of a [Nuimo hub application for Windows IoT Core](https://github.com/wind-rider/nuimo-hub). I started creating the following integrations:
- Philips Hue
- Chromecast
- Timer (idea is to set the timer to a time with the dial ring and then press the button to start counting down)
- A test app to fiddle around with event handling and showing Led Matrices

It needs a forked version (PR #16) of the Nuimo SDK (because for example I needed a way to accumulate the rotation events not to flood the program with triggers, and I wanted to try to improve the connection / reconnection stability).

It is not a Windows desktop application as you might have been looking for, but still you might find it interesting. @soffer @vittel @Inrego @NeilvdM @voelpade tagging you here since you showed interest in the NuimoSDK in the past (I assume @larsblumberg gets notifications automatically :) )

**Comment by hansmbakker** [comment]
I now have a basic working version, but there are still lots of improvements to make.

**Comment by Inrego** [comment]
I have been developing a service to integrate Nuimo into Home Assistant on Windows. It handles integrations with all other devices I'd need :)

**Comment by soffer** [comment]
@wind-rider Thank you for the update.
I'm so frustrated with the Nuimo staff that I simply neglected the thing/
@Inrego Could you also share your project? 

I'm hoping next week to have some time/energy and continue this...

**Comment by grunskis** [comment]
@Inrego Yes, please share your project. I'm also really interested to see how are you using Home Assistant with Nuimo

**Comment by Inrego** [comment]
I will later when I get home.
Note that it is not a finished or even working product yet.

But basically, it is a Windows Service that communicates with the Home Assistant api.
So it doesn't integrate directly into Home Assistant.. Instead you configure actions in a GUI for the service.

Here's a rather old screenshot, but it may give you an idea:
https://twitter.com/Inrego/status/792829353151332352

**Comment by hansmbakker** [comment]
Also, I asked Senic whether they can make a Nuimo firmware that will make it behave as a Surface Dial. This way Nuimo would profit from the integration groundwork that Microsoft laid - saving them time and having a more direct integration in windows (not needing a separate service, app or even a driver).

Microsoft documented how to do this at https://msdn.microsoft.com/en-us/windows/hardware/commercialize/design/component-guidelines/radial-controller-protocol-implementation. The minimum implementation - rotation and button - seems quite straightforward to do.

**Comment by Inrego** [comment]
From what I understand, Senic is working on a kind of bridge device that runs Home Assistant and can connect to Nuimo. I don't know about you, but that would kind of fill my requirements and I wouldn't need to put any more work into development for Nuimo 😛 

I had a Skype call with one of the Senic designers, where he asked me questions about what I'd like from such a device (for example I suggested it could be a Google Home-like device that takes voice commands too).

**Comment by soffer** [comment]
I'm at the stage where I simply ask for Win integration as was promised and as is already operational on macOS...  :-(

**Comment by SeidChr** [comment]
With all these bt libraries for Windows and the fact, that the os level bt integration is only implemented for universal apps that can't be used as drivers in any way, and bt not yet part of .Net standard libraries, makes a platform support hard to realize.

**Comment by Inrego** [comment]
I created a pull request a long time ago which makes this library supported on desktop .NET
Windows 10 or higher is still a requirement, though.

That is also how I integrate Nuimo into a Windows Service in my own development.

**Comment by SeidChr** [comment]
I think i integrated it in my own library version. At least i made it work on desktop. But what i did was so much of a hack that it was the reason why i never pushed the api to github ;)

**Comment by craigdent** [comment]
Did anyone ever get the nuimo to work with home assistant or homebridge using the nuimo connected to the hub?

**Comment by Inrego** [comment]
After spending more time with the device, and being beta tester of their hub - I kinda dropped using it. I don't like that I have to tap the device to wake it up, it cannot wake up by turning the wheel. It's so unintuitive. And with the hub being such a flop, I never really used any of them again. Now I have Google Home and just voice control everything instead.

**Comment by soffer** [comment]
Same here

[Adi Soffer - Contact Using Spike](https://www.spikenow.com/?ref=spike-organic-signature&_ts=prsyd4)

On May 20, 2019 at 11:29 GMT, René Simonsen <notifications@github.com> wrote:

After spending more time with the device, and being beta tester of their hub - I kinda dropped using it. I don't like that I have to tap the device to wake it up, it cannot wake up by turning the wheel. It's so unintuitive. And with the hub being such a flop, I never really used any of them again. Now I have Google Home and just voice control everything instead.
